First commits

jsf9k opened the pull request:
## 🗣 Description ##

This pull request consists of the first commits for this repository.

## 💭 Motivation and context ##

We want to have a Linux instance share the EFS share via Samba so that Windows instances can mount it.  A step toward that is creating a Linux AMI with Samba installed.

## 🧪 Testing ##

All GitHub Actions pass.  I have built an AMI for our staging COOL AMI with these changes and verified that it functions as expected.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.
- [x] Build and test a staging AMI from this repository.

## ✅ Pre-merge checklist ##

- [x] Revert to the main branch of [cisagov/ansible-role-samba](https://github.com/cisagov/ansible-role-samba) once cisagov/ansible-role-samba#2 is merged.
- [x] Revert to the main branch of [cisagov/ansible-role-amazon-efs-utils](https://github.com/cisagov/ansible-role-amazon-efs-utils) once cisagov/ansible-role-amazon-efs-utils#24 is merged.
- [x] Finalize version.

jsf9k committed:
Whittle down the list of codeowners to only folks who know about this repo

jsf9k committed:
Update README for this repo

jsf9k committed:
Rename AMI

jsf9k committed:
Add Samba installation

jsf9k committed:
Add import of Samba playbook in main playbook

jsf9k committed:
Bump version from 0.1.2 to 1.0.0

jsf9k committed:
Bump version from 1.0.0 to 1.0.0-rc.1

jsf9k committed:
Add tasks to configure Samba and its unit file

Also add a role parameter to create the Samba guest user.

jsf9k committed:
Use the ini_file module to modify configuration files

This is much simpler.

jsf9k committed:
Correct incorrect Ansible module name

felddy commented on .github/CODEOWNERS:
```diff
@@ -3,7 +3,7 @@
 # repo. Unless a later match takes precedence, these owners will be
 # requested for review when someone opens a pull request.
-* @dav3r @felddy @jsf9k @mcdonnnj
+* @dav3r @jsf9k @mcdonnnj
```
:(

jsf9k replied:
:violin:

felddy commented on src/samba.yml:
```diff
@@ -0,0 +1,51 @@
+            - {option: "browseable", value: "yes"}
+            - {option: "guest ok", value: "yes"}
+            - {option: "guest only", value: "yes"}
+            - {option: "map to guest", value: "Bad User"}
```
I'm not sure this warrants a comment, but this seemed like a magic string to me.  Found my answer in the [`smb.conf` man page](https://www.samba.org/samba/docs/current/man-html/smb.conf.5.html): 

```
This parameter can take four different values, which tell smbd(8) what to do with user login requests that 
don't match a valid UNIX user in some way... 

Bad User - Means user logins with an invalid password are rejected, unless the username does not exist,
in which case it is treated as a guest login and mapped into the guest account.
```

dav3r replied:
In my opinion, that's interesting enough for a comment.

jsf9k replied:
Please see commit db35b07.

felddy reviewed (approved):
These server message blocks are just fine.  💯

dav3r commented on src/samba.yml:
```diff
@@ -0,0 +1,51 @@
+        create_guest_user: yes
+        server: yes
+  tasks:
+    # This way samba should not start until after the EFS share is
```
```suggestion
    # This way Samba should not start until after the EFS share is
```

dav3r reviewed (approved):
Strong work! 💪 💼

jsf9k committed:
Make capitalization of "Samba" consistent

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k committed:
Add a comment pointing the curious to the smb.conf man page

Some of the smb.conf option, value pairs can be a bit unclear.

Co-authored-by: Mark Feldhousen <felddy@users.noreply.github.com>
Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k committed:
Correct Application tag value

jsf9k committed:
Update terraform-post-packer code from skeleton

jsf9k committed:
Bump version from 1.0.0-rc.1 to 1.0.0-rc.2

jsf9k committed:
Bump version from 1.0.0-rc.2 to 1.0.0-rc.3

jsf9k committed:
Move two options to the global section

jsf9k committed:
Use a particular branch of cisagov/ansible-role-amazon-efs-utils for testing

jsf9k committed:
Specify the GiD for the efs_users group

jsf9k committed:
Specify the UID and additional groups for the Samba guest user

jsf9k committed:
Use the default branch of cisagov/ansible-role-samba

The branch we were using was merged via cisagov/ansible-role-samba#2.

jsf9k committed:
Use the default branch of cisagov/ansible-role-amazon-efs-utils

 The branch we were using was merged via cisagov/ansible-role-amazon-efs-utils#24.

mcdonnnj reviewed (approved):
Makes sense. Is guest access using ACL/SG configuration to ensure access control the long-term strategy?

jsf9k commented:
> Makes sense. Is guest access using ACL/SG configuration to ensure access control the long-term strategy?

I was thinking so, although I hadn't fully decided on it.  The EFS mounts are _just there_ on the instance types that have them, and I don't see how guest access on the SMB share from allowed hosts is any less secure than that.  I was also hoping to avoid the headache of distributing per-assessment SMB share passwords, although I can easily generate them via Ansible [just like I do for the Bloodhound Neo4j password](https://github.com/cisagov/ansible-role-bloodhound/blob/develop/defaults/main.yml#L4-L5).

What are you thinking?

mcdonnnj commented:
> > Makes sense. Is guest access using ACL/SG configuration to ensure access control the long-term strategy?
> 
> I was thinking so, although I hadn't fully decided on it. The EFS mounts are _just there_ on the instance types that have them, and I don't see how guest access on the SMB share from allowed hosts is any less secure than that. I was also hoping to avoid the headache of distributing per-assessment SMB share passwords, although I can easily generate them via Ansible [just like I do for the Bloodhound Neo4j password](https://github.com/cisagov/ansible-role-bloodhound/blob/develop/defaults/main.yml#L4-L5).
> 
> What are you thinking?

That works for me. I wanted to double-check that it was the path we were taking and make sure we had a good foundation for why it was ok to go that way in case it came up in a security review.

Also if you intended to switch to per-assessment passwords that would be something I would have to take into consideration for some of my work x)

jsf9k committed:
Bump version from 1.0.0-rc.3 to 1.0.0
